network/retrieval/retrieve.go

```diff
@@ -386,9 +387,11 @@ func (r *Retrieval) handleChunkDelivery(ctx context.Context, p *Peer, msg *Chunk
 	if err != nil {
 		p.logger.Error("netstore error putting chunk to localstore", "err", err)
 		if err == storage.ErrChunkInvalid {
-			p.Drop("invalid chunk in netstore put")
+			return fmt.Errorf("netstore error putting chunk to localstore: %s", err)
```

Comment:
if the chunk is invalid then we should definitely drop the peer, but if netstore `Put` fails we should definitely consider propagating the error too. @janos? @zelig?